File: DAIDALUS/C++/src/FixedAircraftUrgencyStrategy.cpp
/*
 * FixedAircraftUrgencyStrategy.cpp
 *
 * Most urgent aircraft strategy where the aircraft is fixed
 *
 */

#include "FixedAircraftUrgencyStrategy.h"
#include "KinematicBandsCore.h"

namespace larcfm {

FixedAircraftUrgencyStrategy::FixedAircraftUrgencyStrategy() {
  ac = TrafficState::INVALID.getId();
}

FixedAircraftUrgencyStrategy::FixedAircraftUrgencyStrategy(const std::string& id) {
  ac = id;
}

std::string FixedAircraftUrgencyStrategy::getFixedAircraftId() const {
  return ac;
}

void FixedAircraftUrgencyStrategy::setFixedAircraftId(const std::string& id) {
  ac = id;
}

TrafficState FixedAircraftUrgencyStrategy::mostUrgentAircraft(Detection3D* detector, const OwnshipState& ownship, const std::vector<TrafficState>& traffic, double T) {
  return TrafficState::getTraffic(traffic,ac);
}

UrgencyStrategy* FixedAircraftUrgencyStrategy::copy() const {
  return new FixedAircraftUrgencyStrategy(ac);
}

}

File: DAIDALUS/Java/src/gov/nasa/larcfm/ACCoRD/FixedAircraftUrgencyStrategy.java
package gov.nasa.larcfm.ACCoRD;

import java.util.List;

/* 
 * Most urgent aircraft strategy where the aircraft is fixed
 */
public class FixedAircraftUrgencyStrategy implements UrgencyStrategy {

  private String ac;
  
  public FixedAircraftUrgencyStrategy() {
    ac = TrafficState.INVALID.getId();
  }

  public FixedAircraftUrgencyStrategy(String id) {
    ac = id;
  }
  
  public String getFixedAircraftId() {
    return ac;
  }
  
  public void setFixedAircraftId(String id) {
    ac = id;
  }
  
  public TrafficState mostUrgentAircraft(Detection3D detector, OwnshipState ownship, List<TrafficState> traffic, double T) {
    return TrafficState.getTraffic(traffic,ac);
  }
  
  public UrgencyStrategy copy() {
    return new FixedAircraftUrgencyStrategy(ac);
  }


}
